blocks/header/header.js

```diff
@@ -85,6 +89,132 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+async function fetchSearchData() {
+  if (!window.searchData) {
+    const resp = await fetch(`/${window.location.pathname.split('/')[1]}/query-index.json`);
+    if (resp.ok) {
+      const json = await resp.json();
+      if (json.data) {
+        window.searchData = json;
+      } else {
+        throw new Error('Fetching search data returned unknown format');
+      }
+    } else {
+      throw new Error(`Fetching search data failed with: ${resp.status}`);
+    }
+  }
+  return window.searchData;
+}
+
+async function search(value) {
+  const searchData = await fetchSearchData();
+  const hits = searchData.data
+    .filter((e) => `${e.title} ${e.description}`.toLowerCase().includes(value.toLowerCase()));
+  return hits;
+}
+
+async function searchInput(event) {
+  const { value, aside } = event.target;
+
+  aside.innerHTML = '';
+
+  const url = new URL(window.location);
+  if (value.length >= 3) {
+    url.searchParams.set('ee_search_query', value);
+  } else {
+    url.searchParams.delete('ee_search_query');
+  }
+
+  if (value.length >= 3) {
+    const title = document.createElement('h1');
+    title.classList.add('nav-search-result-title');
+    title.textContent = `Search Results for: ${value}`;
+    aside.append(title);
+    aside.insertAdjacentHTML('beforeend', '<div class="nav-search-result-title-divider"><span class="nav-search-result-title-divider-separator"/></div>');
+
+    try {
+      const hits = await search(value);
+      if (hits.length > 0) {
+        hits.forEach((hit) => {
+          const wrapper = document.createElement('div');
+          wrapper.classList.add('nav-search-wrapper');
+          wrapper.innerHTML = `<h3 class='nav-search-title'><a href='${hit.path}'>${hit.title}</a></h3>
```

Comment:
i think here we have the same issue with user controlled text (author via word document) as a potential security issue, WDYT?

Reply:
Yes, I agree - I guess it is better to be on the safe side. I changed it to be using textContent as well.